src/messages_queue.h

```diff
@@ -150,7 +153,9 @@ class MessagesQueue final {
   void StoreMessages(std::string const & messages) {
     if (current_file_) {
       *current_file_ << messages << std::flush;
-      if (current_file_->tellp() >= kMaxFileSizeInBytes) {
+      if (current_file_->fail()) {
+        ALOG("CRITICAL ERROR: Write to", storage_directory_ + kCurrentFileName, "has failed.");
```

Comment:
Не может - в коде ProcessInitializeStorageCommand, через который меняется storage_directory_, меняется и current_file_

> On Apr 26, 2016, at 11:40, mpimenov <notifications@github.com> wrote:
> 
> In src/messages_queue.h:
> 
> > @@ -150,7 +153,9 @@ class MessagesQueue final {
> >    void StoreMessages(std::string const & messages) {
> >      if (current_file_) {
> >        *current_file_ << messages << std::flush;
> > -      if (current_file_->tellp() >= kMaxFileSizeInBytes) {
> > +      if (current_file_->fail()) {
> > +        ALOG("CRITICAL ERROR: Write to", storage_directory_ + kCurrentFileName, "has failed.");
> 
> Не может возникнуть ситуации, когда current_file -- это не тот же самый файл, что (storage_directory_ + kCurrentFileName)? Т.е. storage_directory_ поменялась, например?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>